src/PublicKey.cpp

```
@@ -56,79 +52,38 @@ PublicKey::PublicKey(const Data& data, enum TWPublicKeyType type)
     if (!isValid(data, type)) {
         throw std::invalid_argument("Invalid public key data");
     }
-    switch (type) {
-    case TWPublicKeyTypeStarkex:
-    case TWPublicKeyTypeSECP256k1:
-    case TWPublicKeyTypeNIST256p1:
-    case TWPublicKeyTypeSECP256k1Extended:
-    case TWPublicKeyTypeNIST256p1Extended:
-        bytes.reserve(data.size());
-        std::copy(std::begin(data), std::end(data), std::back_inserter(bytes));
-        break;
-
-    case TWPublicKeyTypeED25519:
-    case TWPublicKeyTypeCURVE25519:
-        bytes.reserve(ed25519Size);
-        if (data.size() == ed25519Size + 1) {
-            std::copy(std::begin(data) + 1, std::end(data), std::back_inserter(bytes));
-        } else {
-            std::copy(std::begin(data), std::end(data), std::back_inserter(bytes));
-        }
-        break;
-    case TWPublicKeyTypeED25519Blake2b:
-        bytes.reserve(ed25519Size);
-        assert(data.size() == ed25519Size); // ensured by isValid() above
-        std::copy(std::begin(data), std::end(data), std::back_inserter(bytes));
-        break;
-    case TWPublicKeyTypeED25519Cardano:
-        bytes.reserve(cardanoKeySize);
-        std::copy(std::begin(data), std::end(data), std::back_inserter(bytes));
+    auto* pubkey = Rust::tw_public_key_create_with_data(data.data(), data.size(), static_cast<uint32_t>(type));
+    if (pubkey != nullptr) {
+        _impl = Rust::wrapTWPublicKey(pubkey);
+        Rust::CByteArrayWrapper pubkeyData = Rust::tw_public_key_data(_impl.get());
+        bytes = pubkeyData.data;
+    } else {
+        _impl = nullptr;
+        bytes = data;
```

Please throw an exception if couldn't create the PublicKey